cuebot/src/main/java/com/imageworks/spcue/dispatcher/HostReportHandler.java

```diff
@@ -384,6 +386,25 @@ private void changeNimbyState(DispatchHost host, RenderHost rh) {
         }
     }
 
+    /**
+     * Changes the Lock state of the host. Looks at the number of locked cores and sets host to
+     * locked if all cores are locked.
+     *
+     * @param host DispatchHost
+     * @param renderHost RenderHost
+     */
+    private void changeLockState(DispatchHost host, CoreDetail coreInfo) {
+        if (host.lockState == LockState.LOCKED) {
+            if (coreInfo.getLockedCores() < coreInfo.getTotalCores()) {
+                host.lockState = LockState.OPEN;
+                hostManager.setHostLock(host, LockState.OPEN, new Source("cores"));
+            }
+        } else if (coreInfo.getLockedCores() == coreInfo.getTotalCores()) {
```

Comment:
Slightly safer to check `coreInfo.getLockedCores() >= coreInfo.getTotalCores()` (rather than `==`)?

Reply:
Done.